jbpm-designer-client/src/main/java/org/jbpm/designer/client/shared/AssignmentData.java

```diff
@@ -351,6 +356,68 @@ protected void setDisallowedPropertyNames(String disallowedPropertyNames) {
         }
     }
 
+    public Map<String, List<String>> getCustomAssignmentProperties() {
+        return mapCustomAssignmentProperties;
+    }
+
+    protected void setCustomAssignmentProperties(final String customassignmentproperties) {
+        this.mapCustomAssignmentProperties.clear();
+        if (customassignmentproperties != null && !customassignmentproperties.isEmpty()) {
+            String[] caps = customassignmentproperties.split(",");
+            for (String cap : caps) {
+                if (!cap.isEmpty()) {
+                    String[] capParts = cap.split(":");
+                    if (capParts.length > 0 && !capParts[0].isEmpty()) {
+                        String propName = capParts[0].trim();
+                        List<String> listPropValues = new ArrayList<String>();
+                        if (capParts.length > 1) {
+                            String propValues = capParts[1].trim();
+                            String[] pvs = propValues.split(";");
+                            if (pvs.length > 0) {
+                                for (String pv : pvs) {
+                                    pv = pv.trim();
+                                    if (!pv.isEmpty()) {
+                                        listPropValues.add(pv);
+                                    }
+                                }
+                            }
+                        }
+                        this.mapCustomAssignmentProperties.put(propName,
+                                                               listPropValues);
+                    }
+                }
+            }
+        }
+    }
```

Comment:
@hasys we could use streams but it would maybe just reduce couple lines. The issue is that this code is parsing the string values that have multiple separators and it just needs to do all those checks unfortunately imo.